core/src/main/scala/org/ensime/indexer/SearchService.scala

```diff
@@ -163,26 +171,31 @@ class SearchService(
     iwork.flatMap { _ => dwork }
   }
 
-  def extractSymbolsFromClassOrJar(file: FileObject): Future[List[FqnSymbol]] = file match {
-    case classfile if classfile.getName.getExtension == "class" =>
-      // too noisy to log
-      val check = FileCheck(classfile)
-      Future {
-        blocking {
-          try extractSymbols(classfile, classfile)
-          finally classfile.close()
-        }
-      }
-    case jar =>
-      log.debug(s"indexing $jar")
-      val check = FileCheck(jar)
-      Future {
-        blocking {
-          val vJar = vfs.vjar(jar)
-          try scan(vJar) flatMap (extractSymbols(jar, _))
-          finally vfs.nuke(vJar)
+  // this method leak semaphore on every call, which must be released
+  // when the List[FqnSymbol] has been processed (even if it is empty)
+  def extractSymbolsFromClassOrJar(file: FileObject): Future[List[FqnSymbol]] = {
+    def global: ExecutionContext = null
```

Comment:
(it's not strictly necessary, but it means the compiler errors if I forget to pass an explicit EC to some of this I/O code)